### PR TITLE
Laser quantum

### PR DESCRIPTION
**Pull request by takuya-ulm**
## Description

- Bug fix
- Introduce VISA class and Command class

## Motivation and Context

- Some bugs were found in transition to the new core
- Even though the visa package was used in several laser hardware, it was not encapsulated

## How Has This Been Tested?

- Tested with gem 532 with the following psu.
*smd 6000*
*smd 12*
*MPC 6000* 

## Types of changes

-  Bug fix
- ~~New feature~~
- ~~Breaking change (Causes existing functionality to not work as expected)~~

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have documented my changes in `/docs/changelog.md`.
- [ ] My change requires additional/updated documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added/updated the config example for any module docstrings as necessary.
- [x] I have checked that the change does not contain obvious errors
(syntax, indentation, mutable default values, etc.).
- [ ] I have tested my changes using 'Load all modules' on the default dummy configuration.
- [ ] All changed Jupyter notebooks have been stripped of their output cells.

### CODE REVIEW

**Review by LukePiewalker — CHANGES_REQUESTED** [review]
Was testing it with an SMD6000 and found some things which need some adjustment. However we should also test with the other PSU Types if we have some at hand?

File: src/qudi/hardware/laser/laserquantum_laser.py [hunk 1]
```diff
@@ -222,46 +205,28 @@ def set_shutter_state(self, state):
...
     def get_lcd(self):
         """ Get the text displayed on the PSU display.
 
         @return str: text on power supply display
         """
-        if self.psu in (PSUTypes.SMD12, PSUTypes.SMD6000):
-            return ''
-        else:
-            return self.inst.query('STATUSLCD?')
+        return self.cmd.get_lcd_status()
```
[the remark this answers is not in the document]
Comment (takuya-ulm) [reply src/qudi/hardware/laser/laserquantum_laser.py]: 'STATUSLCD' is not implemented, which gives what is written in the display. This is now modified such that it gives empty strings if the device does not have this comman.

File: src/qudi/hardware/laser/laserquantum_laser.py [hunk 2]
```diff
@@ -276,68 +241,199 @@ def set_laser_state(self, status):
...
-            self.inst.write('SOFTVER?')
-        lines = []
-        try:
-            while True:
-                lines.append(self.inst.read())
-        except:
-            pass
-        return lines
+        json.dumps(asdict(self.cmd.sys_info))
+        return
```
[the remark this answers is not in the document]
Comment (takuya-ulm) [reply src/qudi/hardware/laser/laserquantum_laser.py]: Now simoly str is used to convert the dictionary type to the string type. 
It is strange that nothing is obtained at your device. At least, it should give information which is implemented on your PSU. I will double check them.

[the remark this answers is not in the document]
Comment (takuya-ulm) [reply src/qudi/hardware/laser/laserquantum_laser.py]: The problem on no ouput from get_extra_info is also fixed now.

File: docs/changelog.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Changelog
+
+## Pre-Release
+
+### Breaking Changes
+None
```
[the remark this answers is not in the document]
Comment (takuya-ulm) [reply docs/changelog.md:7]: Main is merged into this branch and the changelog is updated.

File: src/qudi/hardware/laser/laserquantum_laser.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TobiasSpohn) [reply src/qudi/hardware/laser/laserquantum_laser.py]: Tested it with SMD6000, works.

File: src/qudi/hardware/laser/laserquantum_laser.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (TobiasSpohn) [reply src/qudi/hardware/laser/laserquantum_laser.py]: Tested it with SMD6000, works.

File: docs/changelog.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (TobiasSpohn) [reply docs/changelog.md]: Merged main again and updated changelog

**Comment by TobiasSpohn** [comment]
This pull looks good from my side. I tested it with SMD6000 for now and it works well.
I made some slight changes:

- fixed default config
- on activation the current power and current setpoint will be set to the current power and current of the device, not just 0
- in the logic, I removed the automatic query loop start upon activation as it is not needed, when only controlling the hardware with it. The GUI will automatically start the query loop upon activation.

**Review by TobiasSpohn — APPROVED** [review]
Thanks Takuya, pull request looks good and was tested by me with SMD6000 and SMD12 PSUs.

**Review by TobiasSpohn — APPROVED** [review]
Changes have been addressed.

**Pull request merged**